File: Graph/Basic/Traversals/DFS/EdgeListDFS/EdgeListDFS.java

import java.util.ArrayList;
import java.util.List;

public class EdgeListDFS {
    static void DFS(int start, int[][] edges, boolean[] visited, List<Integer> dfs) {
        if(!visited[start]) {
            visited[start] = true;
            dfs.add(start);
            for(int[] edge: edges) {
                if(edge[0] == start && !visited[edge[1]]) {
                    DFS(edge[1], edges, visited, dfs);
                } else if(edge[1] == start && !visited[edge[0]]) {
                    DFS(edge[0], edges, visited, dfs);
                }
            }
        }
    }
    static List<Integer> traverse(int V, int[][] edges) {
        List<Integer> dfs = new ArrayList<>();
        boolean[] visited = new boolean[V];
        for(int i=1; i<V; i++) {
            DFS(i, edges, visited, dfs);
        }
        return dfs;
    }
    public static void main(String[] args) {
        int[][] edges = {{2, 1}, {2, 4}, {4, 3}, {1, 3}, {5, 6}, {6, 7}, {5, 7}, {8, 9}};
        List<Integer> list = traverse(11, edges);
        System.out.println(list);
    }
}

File: Graph/Basic/Traversals/DFS/EdgeListDFS/EdgeListDFS.cpp
#include<bits/stdc++.h>
using namespace std;
// original DFS logic
void DFS(int start, vector<pair<int, int>> &edges, vector<int> &visited, vector<int> &dfs) {
    if(!visited[start]) {
        visited[start] = 1;
        dfs.push_back(start);
        for (auto edge : edges)
        {
            if (edge.first == start && visited[edge.second] == 0)
            {
                DFS(edge.second, edges, visited, dfs);
            }
            else if (edge.second == start && visited[edge.first] == 0)
            {
                DFS(edge.first, edges, visited, dfs);
            }
        }
    }
}
vector<int> traversal(int V, vector<pair<int, int>> &edges) {
    vector<int> visited(V, 0);
    vector<int> dfs;
    for(int i=1; i<V; i++) {
        DFS(i, edges, visited, dfs);
    }
    return dfs;
}
int main(void) {
    vector<pair<int, int>> edges;
    edges.push_back(make_pair(2, 1));
    edges.push_back(make_pair(2, 4));
    edges.push_back(make_pair(4, 3));
    edges.push_back(make_pair(1, 3));
    edges.push_back(make_pair(5, 6));
    edges.push_back(make_pair(6, 7));
    edges.push_back(make_pair(5, 7));
    edges.push_back(make_pair(8, 9));
    vector<int> dfs = traversal(11, edges);
    for(int ele: dfs) {
        cout << ele << " ";
    }
    return 0;
} 